### ISSUE TITLE
resizeable datasets: problematic if first extent is zero

### ISSUE THREAD

**Issue by ax3l** [openPMD/openPMD-api#1094]
With resizeable datasets, there is a problem to resize them if we first declare them as "zero" size.
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  Chunks cannot be written for a constant RecordComponent.
SIGABRT
```

Since handling zero-extent datasets is somewhat special (they need to become constant record components), it might be that we want to guard this in the frontend (throw an error if a resizeable data set is declared with zero extent).

The work-around/proper workflow on the user side is to wait with the declaration until N>1 particles are truly to be written.

seen first by @RevathiJambunathan & me in WarpX back-transformed particle diags .

**Comment by franzpoeschel** [comment]
Yeah, declaring a dataset with a zero-size extent (on any dimension) will automatically make it scalar.
Now, extending it to be non-zero [does currently not throw](https://github.com/openPMD/openPMD-api/blob/c3c724f57625cf626ba518cd488e3355f962b243/test/SerialIOTest.cpp#L4346), but the dataset will stay scalar. Since there is no way to overwrite the value of constant record components (or is there?), setting a value other than the type's zero value will not work.

We could make this throw, but it will break currently tested behavior. I would argue that this is not a bug (and the possibility to change a constant value just a missing feature), so I'm not sure we should really put this in a bugfix release?
